packages/perseus/src/widgets/interactive-graphs/graphs/segment.tsx

```diff
@@ -24,12 +28,36 @@ export function renderSegmentGraph(
 
 type SegmentProps = MafsGraphProps<SegmentGraphState>;
 
-const SegmentGraph = (props: SegmentProps) => {
-    const {dispatch} = props;
-    const {coords: segments} = props.graphState;
+const SegmentGraph = ({dispatch, graphState}: SegmentProps) => {
+    const {coords: segments} = graphState;
+    const {strings, locale} = usePerseusI18n();
+    const lengthOfSegment =
+        segments[0] && kpoint.distanceToPoint(...segments[0]);
+
+    const wholeSegmentAriaLabel = strings.srSegmentGraphAriaLabel;
+    const wholeSegmentAriaDescription =
+        segments[0] &&
+        strings.srSegmentGraphAriaDescription({
+            point1X: srFormatNumber(segments[0][0][X], locale),
+            point1Y: srFormatNumber(segments[0][0][Y], locale),
+            point2X: srFormatNumber(segments[0][1][X], locale),
+            point2Y: srFormatNumber(segments[0][1][Y], locale),
+            length: srFormatNumber(lengthOfSegment, locale),
+        });
```

Comment:
EDIT: I know why this is happening. You have to use unique IDs to fix this. You can see how I did this for circle and linear if you need a reference.

I'm not really sure what's going on, but it seems like this is somehow using the correct answer or default coordinates instead of the ones that are on the graph? I'm extra confused because your implementation looks right to me... 🤔 

It has the right values for the points themselves, just not for the aria describedby.

https://github.com/user-attachments/assets/7977804d-3853-431e-87dc-028b5952292e